common/js_handle.go

```diff
@@ -168,8 +163,25 @@ func (h *BaseJSHandle) GetProperty(_ string) JSHandleAPI {
 }
 
 // JSONValue returns a JSON version of this JS handle.
-func (h *BaseJSHandle) JSONValue() goja.Value {
-	if h.remoteObject.ObjectID != "" {
+func (h *BaseJSHandle) JSONValue() string { // TODO: return error
+	convertToString := func(ro *cdpruntime.RemoteObject) string {
+		res, err := valueFromRemoteObject(h.ctx, ro)
+		if err != nil {
+			k6ext.Panic(h.ctx, "extracting value from remote object: %w", err)
+		}
+		switch v := res.(type) {
+		case string:
+			return v
+		default:
+			buf, err := json.Marshal(v)
+			if err != nil {
+				k6ext.Panic(h.ctx, "marshaling value: %w", err)
+			}
+			return string(buf)
```

Comment:
This is for ensuring `JSONValue` returns a string representation of the CDP/JS object returned from `valueFromRemoteObject`.

Reply:
Added as a task in #1182. Let's reconsider it there 👍